src/aggregated_queries.rs

```diff
@@ -19,63 +18,96 @@ pub struct AggregatedQueriesPerLanguage {
     pub query: Arc<Query>,
     #[allow(dead_code)]
     pub query_text: String,
+    pub kind_exit_rule_listener_indices: HashMap<String, Vec<(RuleIndex, RuleListenerIndex)>>,
 }
 
 #[derive(Debug, Default)]
 struct AggregatedQueriesPerLanguageBuilder {
     pattern_index_lookup: Vec<(RuleIndex, RuleListenerIndex, CaptureNameIfPerCapture)>,
     query_text: String,
+    kind_exit_rule_listener_indices: HashMap<String, Vec<(RuleIndex, RuleListenerIndex)>>,
 }
 
 impl AggregatedQueriesPerLanguageBuilder {
+    #[instrument(level = "trace")]
     pub fn build(self, language: SupportedLanguage) -> AggregatedQueriesPerLanguage {
         let Self {
             pattern_index_lookup,
-            query_text,
+            mut query_text,
+            kind_exit_rule_listener_indices,
         } = self;
+
+        if !kind_exit_rule_listener_indices.is_empty() {
+            query_text.push_str(&format!(
+                "\n[{}] @c",
```

Comment:
Tried to optimize this a little by making it only match node kinds that people actually asked for exit listeners for

I guess that will get "blown away" by event emitters though which presumably have to get "the firehose"?